internal/terraform/evaluate.go

```diff
@@ -989,10 +989,10 @@ func (d *evaluationStateData) GetOutput(addr addrs.OutputValue, rng tfdiags.Sour
 			Value:     cty.NilVal,
 			Sensitive: config.Sensitive,
 		}
-	} else if output.Value == cty.NilVal {
+	} else if output.Value == cty.NilVal || output.Value.IsNull() {
 		// Then we did get a value but Terraform itself thought it was NilVal
 		// so we treat this as if the value isn't yet known.
-		output.Value = cty.DynamicVal
+		output.Value = cty.UnknownVal(cty.DynamicPseudoType)
```

Comment:
Was this just some churn or was there an intended change here? Only asking because `DymamicVal` is `UnknownVal(cty.DynamicPseudoType)`

Reply:
Just churn, I just wanted the two places we parse the outputs from run blocks to be the same and I settled on the second choice. I figured they were functionally identical anyway. Happy to keep changes to a minimum so reverted this, and used cty.DynamicVal in the other place as well.